src/compiler/Stage/Parse/Parser.elm

```diff
@@ -660,6 +661,20 @@ unit _ =
         |> P.inContext InUnit
 
 
+list : ExprConfig -> Parser_ Frontend.Expr
+list config =
+    P.succeed Frontend.List
+        |= P.sequence
+            { start = P.Token "[" ExpectingLeftBracket
+            , separator = P.Token "," ExpectingListSeparator
+            , end = P.Token "]" ExpectingRightBracket
+            , spaces = spacesOnly
+            , item = PP.subExpression 0 config
+            , trailing = P.Forbidden
+            }
+        |> P.inContext InList
```

Comment:
Cool to see `P.sequence` works well. I'm a bit careful regarding those helpers :sweat_smile:

Reply:
It's not perfect, but I guess it was specifically design with that usecase in mind, so I'm confident it can stay like this until perfs become a thing.